src/lando/api/legacy/workers/base.py

```diff
@@ -6,21 +6,62 @@
 import os
 import re
 import subprocess
+from abc import ABC, abstractmethod
 from time import sleep
-from typing import Optional
+
+from celery import Task
+from django.db import transaction
+from kombu.exceptions import OperationalError
 
 import lando.utils.treestatus
+from lando.api.legacy.treestatus import TreeStatus
+from lando.main.models import (
+    BaseJob,
+    JobStatus,
+    Repo,
+    WorkerType,
+)
 from lando.main.models import Worker as WorkerModel
 
 logger = logging.getLogger(__name__)
 
 
-class Worker:
+class Worker(ABC):
     """A base class for repository workers."""
 
     SSH_PRIVATE_KEY_ENV_KEY = "SSH_PRIVATE_KEY"
 
-    ssh_private_key: Optional[str]
+    @property
+    @abstractmethod
+    def job_type(self) -> type[BaseJob]:
+        """Type of this Job that this worker can process.
+
+        This should be overridden by concrete implementations."""
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def type(self) -> WorkerType:
+        """Type of this Worker.
+
+        This should be overridden by concrete implementations."""
+        raise NotImplementedError()
```

Comment:
I would get rid of these methods in favour of setting the variables on the subclasses themslves. Also, should avoid using built-in function names as method names.

```suggestion
    job_type: LandingJob | AutomationJob
    worker_type: LandingWorker | AutomationWorker
```

Though for both of these variables, I would suggest automatically detecting that based on a class name mapping, since that information is available and doesn't need to be explicitly set.